packages/http-client/src/http-client-types.ts

```diff
@@ -77,3 +88,29 @@ export interface HttpRequestOptions {
    */
   fetchCsrfToken?: boolean;
 }
+
+/**
+ * Origins of http request options. This indicates the priority of an http request option.
+ * Http request options with higher priorities will be used when reaching conflicts.
+ * The priority is "Custom" \> "DestinationProperty" \> "Destination" \> "RequestConfig"
+ */
+export type Origin =
+  | 'Custom'
+  | 'DestinationProperty'
+  | 'Destination'
+  | 'RequestConfig';
+
+/**
+ * @internal
+ */
+export type ValueWithOrigin = {
+  [key in Origin]?: string;
+};
+
+/**
+ * @internal
+ */
+export interface OptionWithOrigin {
```

Comment:
I find this a bit misleading. It is more Options from origin right? So all header or parameter from one origin.